types/utils.go

```diff
@@ -71,92 +57,6 @@ func isRootlessRuntimeDirOwner(dir string, env rootlessRuntimeDirEnvironment) bo
 	return err == nil && int(st.UID()) == os.Getuid() && st.Mode()&0o700 == 0o700 && st.Mode()&0o066 == 0o000
 }
 
-// getRootlessRuntimeDirIsolated is an internal implementation detail of getRootlessRuntimeDir to allow testing.
-// Everyone but the tests this is intended for should only call getRootlessRuntimeDir, never this function.
-func getRootlessRuntimeDirIsolated(env rootlessRuntimeDirEnvironment) (string, error) {
```

Comment:
I’d prefer adapting the `rootlessRuntimeDirEnvironment`-based implementation… but if it were to be removed, then all of that interface and the default implementation should be removed as well.